### PR TITLE
Endpoint testing with mocks.

### PR DESCRIPTION
**Pull request by AlexanderVladimirov9090**

### CODE REVIEW

File: test/com/clouway/endpointfilter/StartsWithKeywordTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * @author alexandervladimirov1902@gmail.com
+ *         (Alexander Vladimirov)
+ */
+public class StartsWithKeywordTest extends EndpointFilterTest {
+
+  @Test
+  public void happyPath() throws EmptyURLExceptions, EmptyKeywordException {
+    StartsWithKeyword startsWithKeyword = new StartsWithKeyword("Start");
+
+    EndpointFilter endpointFilter = new EndpointFilter(startsWithKeyword);
```
Comment (mgenov) [inline test/com/clouway/endpointfilter/StartsWithKeywordTest.java]: EndpointFilter should not be used for testing of StartsWithKeyword, cause it leads to integrated test.

**Comment by AlexanderVladimirov9090** [comment]
Yeah will fix it now.

File: test/com/clouway/endpointfilter/EndpointFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Created by clouway on 04.07.16.
+ *
+ * @author alexandervladimirov1902@gmail.com
+ *         (Alexander Vladimirov)
+ */
+public class EndpointFilterTest {
```
Comment (mgenov) [inline test/com/clouway/endpointfilter/EndpointFilterTest.java]: You have to verify and the case when:

1 endpoint  is not matching
2 endpoint is not matching
3 endpoint is matching

This is commonly named as "many" case.

Comment (AlexanderVladimirov9090) [inline test/com/clouway/endpointfilter/EndpointFilterTest.java]: Ou that will be in a sequence. 
Will do it.

**Comment by AlexanderVladimirov9090** [comment]
Done.

File: test/com/clouway/endpointfilter/EndpointFilterTest.java [hunk 3]
```diff
@@ -57,4 +58,28 @@ public void urlDoesNotMatch() throws EmptyURLExceptions, EmptyKeywordException {
...
+      will(returnValue(false));
+      oneOf(endpoint).matches("ThisWillNotMatchAgain");
+      inSequence(match);
+      will(returnValue(false));
+      oneOf(endpoint).matches("ThisWillMatch");
+      inSequence(match);
+      will(returnValue(true));
+    }});
+
+    EndpointFilter endpointFilter = new EndpointFilter(endpoint);
```
Comment (mgenov) [inline test/com/clouway/endpointfilter/EndpointFilterTest.java]: The idea was: EndpointFilter endpointFilter = new EndpointFilter(firstEndpoint, secondEndpoint, thirdEndpoint)

If you change the production code to this: 

```java
public boolean shouldFilter(String url) throws EmptyURLExceptions, EmptyKeywordException {
   return endpoints[0].matches(url);
  }
```

All tests will still pass, but the EndpointFilter will be able to work correctly only with a single endpoint.

**Comment by AlexanderVladimirov9090** [comment]
I have noticed something and will fixed it.

**Pull request closed without merging**